test/e2e/configuration_test.go

```diff
@@ -2,118 +2,138 @@ package e2e_test
-			kube.Apply(userSpec.clusters[0].ClusterFileName(), "-n", userSpec.namespace)
-		})
-
-		By("Wait project creation", func() {
-			waitProject(userSpec, "1")
-			userSpec.projectID = kube.GetProjectResource(userSpec.namespace, userSpec.k8sFullProjectName).Status.ID
-		})
-
-		By("Wait cluster creation", func() {
-			waitCluster(userSpec, "1")
-		})
-
-		By("check cluster Attribute", func() { // TODO ...
-			cluster := mongocli.GetClustersInfo(userSpec.projectID, userSpec.clusters[0].Spec.Name)
-			Expect(
-				cluster.ProviderSettings.InstanceSizeName,
-			).Should(Equal(userSpec.clusters[0].Spec.ProviderSettings.InstanceSizeName))
-			Expect(
-				cluster.ProviderSettings.ProviderName,
-			).Should(Equal(string(userSpec.clusters[0].Spec.ProviderSettings.ProviderName)))
-			Expect(
-				cluster.ProviderSettings.RegionName,
-			).Should(Equal(userSpec.clusters[0].Spec.ProviderSettings.RegionName))
-		})
-
-		By("Update cluster\n", func() {
-			userSpec.clusters[0].Spec.ProviderSettings.InstanceSizeName = "M20"
-			utils.SaveToFile(
-				userSpec.clusters[0].ClusterFileName(),
-				utils.JSONToYAMLConvert(userSpec.clusters[0]),
-			)
-			kube.Apply(userSpec.clusters[0].ClusterFileName(), "-n", userSpec.namespace)
-		})
-
-		By("Wait cluster updating", func() {
-			waitCluster(userSpec, "2")
-		})
-
-		By("Check attributes", func() {
-			uCluster := mongocli.GetClustersInfo(userSpec.projectID, userSpec.clusters[0].Spec.Name)
-			Expect(
-				uCluster.ProviderSettings.InstanceSizeName,
-			).Should(Equal(
-				userSpec.clusters[0].Spec.ProviderSettings.InstanceSizeName,
-			), "Instance size should be the same as requested by the user")
-		})
-
-		By("Delete cluster", func() {
-			kube.Delete(userSpec.clusters[0].ClusterFileName(), "-n", userSpec.namespace)
-			Eventually(
-				checkIfClusterExist(userSpec),
-				"10m", "1m",
-			).Should(BeFalse(), "Cluster should be deleted from Atlas")
-		})
-
-		By("Delete project", func() {
-			kube.Delete(userSpec.projectPath, "-n", userSpec.namespace)
-			Eventually(
-				func() bool {
-					return mongocli.IsProjectInfoExist(userSpec.projectID)
-				},
-				"5m", "20s",
-			).Should(BeFalse(), "Project should be deleted from Atlas")
-		})
-	})
+	newData := func(description, path string) (string, testDataProvider) {
+		var data testDataProvider
+		data.description = description
+		data.confPath = path
+		data.userSpec = NewUserInputs("my-atlas-key")
+		return data.description, data
+	}
+
+	DescribeTable("Namespaced operators working only with its own namespace with different configuration",
+		func(test testDataProvider) {
+			data = test
+			mainCycle(test.confPath, test.userSpec)
+		},
+		Entry(newData("Simple Test with default cluster", "data/atlascluster_basic.yaml")),
+		Entry(newData("Simple Backup Configuration and 2 users", "data/atlascluster_backup.yaml")),
+		// Entry(newData("Multiregion, Backup and 2 users", "data/atlascluster_multiregion.yaml")), // TODO CLOUDP-83419
```

Comment:
can we please create a ticket to uncomment this as soon as the fix for ReplicationSpecs arrives?